src/v/cloud_storage/partition_manifest.cc

```diff
@@ -1420,6 +1465,17 @@ struct partition_manifest_handler
                 return true;
             }
             return false;
+        case state::expect_spillover_meta_value:
+            if (_segment_meta_key == "is_compacted") {
+                _is_compacted = b;
+                _state = state::expect_spillover_meta_key;
+                return true;
+            }
+            return false;
```

Comment:
nit: the `is_compacted` field does not apply to spillover manifests represented via `segment_meta`, so we could always return `false` here.